src/main/java/org/sonarlint/intellij/analysis/SonarLintTask.java

```diff
@@ -51,6 +62,11 @@ public boolean shouldStartInBackground() {
     return startInBackground;
   }
 
+  @Override
+  public boolean isConditionalModal() {
```

Comment:
Check CoreProgressManager:315. For these type of tasks, it will decide if it runs synchronously or asynchronously based on isConditionalModal and shouldStartInBackground.